tracer/src/Datadog.Trace/AppSec/Waf/Encoder.cs

```diff
@@ -56,7 +56,7 @@ public static ObjType DecodeArgsType(DDWAF_OBJ_TYPE t)
         private static string TruncateLongString(string s) =>
             s.Length > WafConstants.MaxStringLength ? s.Substring(0, WafConstants.MaxStringLength) : s;
 
-        public Obj Encode(object o, List<Obj> argCache, bool applySafetyLimits) =>
+        public Obj Encode(object o, List<Obj> argCache = null, bool applySafetyLimits = false) =>
             EncodeInternal(o, argCache, WafConstants.MaxContainerDepth, applySafetyLimits);
```

Comment:
Defaulting `applySafetyLimits` to `false` seems bold to me... Personally I'd prefer they were explicit. Also, if you're going to start passing in `null` to `List<>` arguments, I'd strongly recommend adding `#nullable enable`..

Reply:
I've defaulted it to false because I saw it was the most used value, but I'll have a look at it once again. And you are right on the '#nullable enable'.